DirectProgramming/DPC++FPGA/Tutorials/Features/ac_types/ap_float/README.md

```diff
@@ -0,0 +1,471 @@
+operations use lesser resources for the `ApproximateSineWithAPFloat` kernel.
+
+You should observe an area reduction of up to 30% in resource utilization of the binary operations.
+
+TODO: Is simulation supported for customers yet?
+Open the "Throughput Analysis > Verification Statistics" page. You should observe around 15% percent latency reduction when using the `ap_float` type.
+
+You should also note that simulations indicate that the values computed for both designs are the same. This indicates that our original precision expectation is still satisfied.
+
+### Examining the Reports for Conversion Kernels
+
+You can find the usages of conversion in both the area report and the graph viewer. The name of the rounding block is "cast".
+Let's look at the reports and analyze each kernel in the tutorial.
+
+1. Kernel: `ConversionKernelA`
+  This kernel uses the default rounding mode `RNE` - round to nearest.
+
+  Navigate to the *System Viewer* report (*Views* > *System Viewer*) and on the left pane, click on the cluster under `ConversionKernelA`. You should see the conversion functions mentioned in the comments of the source code as "cast" nodes in the graph. The casts from literal types are eliminated at compile time.
+
+  The 4 cast nodes correspond to the following pieces of code:
+
+  ```cpp
+  const floatTy z = 4.1;
+  ```
+
+  ```cpp
+  auto res = x * num_accessor[0] + y * num_accessor[0] + z;
+  ```
+
+  The comments in the kernel code describe how these 2 lines generate 4 "cast" nodes.
+
+2. Kernel: `ConversionKernelB`
+  This kernel uses the simpler rounding mode `RZERO`.
+
+  In the graph for the cluster under `ConversionKernelB`, you will find that it now only contains one "cast" node. This corresponds to the code:
+  ```cpp
+    x * num_accessor[0] + ...
+  ```
+  Although `x` and `num_accessor[0]` represent ap_floats constructed to use rounding mode `RZERO`, the result of this operation is cast to the higher precision `ap_float` using the default rounding mode `RNE` as the multiplication result is an operand for the next operation which uses higher precision.
+
+  The other cast node is represented by a combination of `shift`, `select`, and `and` operations hence only one node labeled as "cast" is visible in the reports.
+  
+  The reduction in the number of cast nodes as compared to `ConversionKernelA` results in reduction of hardware resources used by `ConversionKernelB`.
+  
+  Observe the differences in the resource usage of these two kernels by navigating to the *Area Analysis of System* report (*Area Analysis* > *Area Analysis of System*) and looking at the entries under `Kernel System`.
+
+2. Kernel: `ConversionKernelC`
+  This kernel shows how to use the `convert_to` function and modify the rounding mode for a specific operation.
+
+  In the graph for the cluster under `Kernel_C`, you will find that it contains two "cast" nodes, corresponding to the conversions:
```

Comment:
does it make sense to include screenshots? probably not since we change the report graphics so much.